_P177_CM1107.ino

```diff
@@ -0,0 +1,235 @@
+//#######################################################################################################
+//######################## Plugin 177 CM1107 I2C CO2 Sensor ############################################
+//#######################################################################################################
+// development version
+// by: V0JT4
+
+
+#define PLUGIN_177
+#define PLUGIN_ID_177         177
+#define PLUGIN_NAME_177       "Gases - CO2 CM1107"
+#define PLUGIN_VALUENAME1_177 "CO2"
+
+boolean Plugin_177_init = false;
+
+
+#define CM1107_ADDR   0x31 // default address
+
+#define  CM1107_CMD_CO2     0x01
+#define  CM1107_CMD_ABC     0x10
+#define  CM1107_CMD_CAL     0x03
+#define  CM1107_CMD_SN      0x1F
+#define  CM1107_CMD_VER     0x1E
+
+#define  CM1107_STATUS_INIT 0x01 //0 1 swapped compared to datasheet
+#define  CM1107_STATUS_OK   0x00
+#define  CM1107_STATUS_ERR  0x02
+#define  CM1107_STATUS_OUT  0x03
+#define  CM1107_STATUS_NCAL 0x05
+
+#define  CM1107_CAL_ON 0x00
+#define  CM1107_CAL_OFF 0x02
+
+
+
+boolean plugin_177_begin()
+{
+  Wire.begin();
+  return(true);
+}
+
+boolean plugin_177_setABC(unsigned char abc)
+  // Set auto calibration period on CM1107
+  // Returns true (1) if successful, false (0) if there was an I2C error
+{
+  // Write auto calibration configuration
+  Wire.beginTransmission(CM1107_ADDR);
```

Comment:
Why not use the functions in the existing I2C.ino file?
This makes it much harder to fix bugs or change I2C related code later.
Also lots of plugins tend to do just the same, so better use the existing functions then it is also clear what's being done.

Reply:
I didn't find any function that would cover sensor's requirements, number of bytes to read and write at once.